### PR TITLE
chore: reference lifecycle

### LINKED ISSUE #37451: [Task] Switch create branch method to super service
Notion Link: https://www.notion.so/Task-Switch-create-branch-method-to-super-service-13cfe271b0e28057963dce2d18033dfe

Notion Content:

### PR DESCRIPTION
**Pull request by sondermanish**
## Description

Fixes #37451, #37455, #37448
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!IMPORTANT]
> 🟣 🟣 🟣 Your tests are running.
> Tests running at: <https://github.com/appsmithorg/appsmith/actions/runs/12370910561>
> Commit: 03a3843d131e4f5d819ede06212ba6300fab99c3
> Workflow: `PR Automation test suite`
> Tags: `@tag.Git`
> Spec: ``
> <hr>Tue, 17 Dec 2024 10:18:10 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for handling Git references and statuses.
	- Added functionality to list branches and references, including options for remote checks.
	- New class `GitRefDTO` created to represent Git references.
	- New enumeration `RefType` defined to categorize reference types.
	- Enhanced Git operations with methods for checking out references and creating new branches.

- **Bug Fixes**
	- Updated error handling and logging in Git status retrieval.

- **Chores**
	- Deprecated the `ORGANIZATION_ID` constant and introduced `REF_NAME`. 
	- Updated import statements for `RefType` across multiple classes.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCE.java [hunk 1]
```diff
@@ -33,4 +34,15 @@ Mono<String> fetchRemoteChanges(
             RefType refType);
 
     Mono<? extends Artifact> discardChanges(String branchedArtifactId, ArtifactType artifactType, GitType gitType);
+
+    Mono<GitStatusDTO> getStatus(
+            String branchedArtifactId, boolean compareRemote, ArtifactType artifactType, GitType gitType);
+
+    Mono<? extends Artifact> checkoutReference(
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCE.java:41]: Nit, should we not be able to make a DTO out of these params?

**Review by nidhi-nair — DISMISSED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCEImpl.java [hunk 2]
```diff
@@ -437,6 +439,134 @@ protected Mono<? extends Artifact> checkoutRemoteBranch(Artifact baseArtifact, S
         return null;
     }
 
+    public Mono<? extends Artifact> checkoutReference(
+            String referencedArtifactId,
+            GitRefDTO refDTO,
+            ArtifactType artifactType,
+            GitType gitType,
+            RefType refType) {
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCEImpl.java:447]: Why are we sending ref type separately here?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCE.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCE.java:41]: Will you be replacing usage in a separate PR?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCEImpl.java (hunk 2 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/git/central/CentralGitServiceCEImpl.java:447]: We can put it in the DTO itself, right.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**